client/src/boot/buildApolloClient.js

```diff
@@ -16,8 +16,8 @@ function buildApolloClient(baseUrl) {
     shouldBatch: true,
     addTypename: true,
     dataIdFromObject: (o) => {
-      if (o.id >= 0 && o.__typename) {
-        return `${o.__typename}:${o.id}`;
+      if ((o.id >= 0 || o.ID >= 0) && o.__typename) {
```

Comment:
Can you explain the case issue here? Should we check that at least one of id or ID are not undefined first?

Maybe shift the ID into a var instead for clarity.